### DISCUSSION TITLE
Support for copying non-code-blocks onto clipboard

### DISCUSSION THREAD

**Discussion by danielrosehill** [mkdocs/mkdocs#3867] [Ideas]
Hi everyone!

I'm working on converting some Github repos into documentation sites and am really enjoying using MK Docs!

One idea I had today:

One of my repos is about prompt engineering. So most of the documentation consists of example prompts and outputs. 

It would be great to be able to provide the user with an easy way to copy the text.

I thought of perhaps using code blocks alongside the code copy plugin and setting the "syntax" to be simply text. 

But it makes the text difficult to read:

![image](https://github.com/user-attachments/assets/fa56c8e7-c963-4538-a5fd-0efb7bc29ce3)

I can use block quotes of course and the text looks much better. But then I don't have the nice 'copy' function that I do if I set it up as a code block.

![image](https://github.com/user-attachments/assets/c6d16397-e6dc-4832-9c9d-edd724e295ff)

It would be great to have a way to mark a block quote for copying or perhaps even declare it as natural language in the code block syntax to prevent the usual format rules from applying.

**Comment by pawamoy** [comment]
I'd recommend *avoiding* rendering the markup (your prompts look like Markdown), as it could prevent users from copying the actual prompt. Therefore, I recommend using a code block indeed, which will prevent rendering of the markup. If you want to make it more readable, akin to the quote, by wrapping the text, you could try this: https://stackoverflow.com/a/248013/3451029. Just give a CSS class to your prompt code blocks, and use it to wrap their contents.

````md
My prompt code block:

```text {.prompt-block}
blabla
```
````

See https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#injecting-classes-ids-and-attributes.
